module/zfs/vdev_label.c

```diff
@@ -508,6 +508,9 @@ vdev_config_generate(spa_t *spa, vdev_t *vd, boolean_t getstats,
 			fnvlist_add_uint64(nv, ZPOOL_CONFIG_REMOVING,
 			    vd->vdev_removing);
 		}
+		if (vd->vdev_rz_expanding) {
+			fnvlist_add_boolean(nv, ZPOOL_CONFIG_RAIDZ_EXPANDING);
```

Comment:
I'm trying to understand why this change is needed.  I see that in spa_vdev_attach(), we set `vdev_rz_expanding` and then do the vdev_raidz_attach_sync synctask, which will set vre_state to DSS_SCANNING, and dirty the config, causing a new config to be written this txg with ZPOOL_CONFIG_RAIDZ_EXPANDING.  The current code should have ZPOOL_CONFIG_RAIDZ_EXPANDING set in the same txg as the changes to `VDEV_TOP_ZAP_RAIDZ_EXPAND_*`.

Since vdev_rz_expanding may be set slightly before this, with the change here, we might set  ZPOOL_CONFIG_RAIDZ_EXPANDING in one txg, and then do vdev_raidz_attach_sync() in a slightly later txg.  Why is that the desired behavior?

Reply:
The idea was to know about expansion starting before the vdev_raidz_attach_sync() will be called, mean
as early as possible. If zfs will crash before vdev_raidz_attach_sync(), we will know that user tried
raidz expansion. From other side in this situation, the expansion will be started after crash without 
vdev_raidz_attach_sync() call, mean without vdev_top_zap fields initialization.

So, it is better to revert this change, to avoid state, when we have ZPOOL_CONFIG_RAIDZ_EXPANDING in the vdev
config, but not VDEV_TOP_ZAP_RAIDZ_EXPAND_* values in the top zap.